lib/fiber/create-instance-from-react-element.ts

```diff
@@ -69,9 +87,23 @@ const hostConfig: HostConfig<
       )
     }
 
-    const instance = prepare(new target(props) as any, {})
+    try {
+      const instance = prepare(new target(props) as any, {})
+      return instance
+    } catch (error) {
+      const rootCircuit = CircuitContext.current
 
-    return instance
+      if (rootCircuit?.db) {
+        // Add the error to the circuit-json database
+        rootCircuit.db.source_failed_to_create_component_error.insert({
+          type: "source_failed_to_create_component_error",
+          component_type: type,
+          message: error instanceof Error ? error.message : String(error),
+          error_type: "source_failed_to_create_component_error",
+        })
+      }
+      return createErrorPlaceholderComponent(type, props, error)
```

Comment:
this is an interesting idea, why not have the placeholder component render the error? We could have it be a Renderable with props, that would allow inspection. It would render the error to circuit json